### PR TITLE
feat: add remote vertex model support

### PR DESCRIPTION
**Pull request by GarrettWu**
b/299356085

### CODE REVIEW

File: bigframes/ml/remote.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+_REMOTE_MODEL_STATUS = "remote_model_status"
+
+
+class VertexAIModel(base.BaseEstimator):
+    """Remote model from a Vertex AI https endpoint. User must specify https endpoint, input schema and output schema.
+    How to deploy a model in Vertex AI https://cloud.google.com/bigquery/docs/bigquery-ml-remote-model-tutorial#Deploy-Model-on-Vertex-AI.
+
+    Args:
+        endpoint (str):
+            Vertex AI https endpoint.
```
Comment (ashleyxuu) [inline bigframes/ml/remote.py:47]: Could you give an example of the endpoints? For example, I saw `"https://us-central1-aiplatform.googleapis.com/v1/projects/1084210331973/locations/us-central1/endpoints/3193318217619603456"` in your test. Might be give users some hints how to find the endpoints?

**Review by ashleyxuu — COMMENTED** [review]
You should also update `docs/template/toc.yml` file and `docs/reference/bigframe.ml` file to auto-sync the docs changes.

File: bigframes/ml/remote.py (hunk 1 above)
Comment (GarrettWu) [reply bigframes/ml/remote.py:47]: User needs to deploy the model by themselves then they can find the endpoint, as well as the input/output schema. Such as https://screenshot.googleplex.com/49NX9PYEbBFNeMH

**Comment by ashleyxuu** [comment]
Could you also add the log_adapter annotation to the new Class?

**Comment by GarrettWu** [comment]
> You should also update `docs/template/toc.yml` file and `docs/reference/bigframe.ml` file to auto-sync the docs changes.

Done.

**Comment by GarrettWu** [comment]
> Could you also add the log_adapter annotation to the new Class?

done

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**